File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/gui/hotbar/HotbarListener.java
package com.gmail.borlandlp.minigamesdtools.arena.gui.hotbar;

import com.gmail.borlandlp.minigamesdtools.MinigamesDTools;
import com.gmail.borlandlp.minigamesdtools.arena.ArenaEventListener;
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaEventHandler;
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaPlayerJoinLocalEvent;
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaPlayerLeaveLocalEvent;
import org.bukkit.event.Listener;

public class HotbarListener implements Listener, ArenaEventListener {
    private HotbarController hotbarController;

    public HotbarListener(HotbarController controller) {
        this.hotbarController = controller;
    }

    @ArenaEventHandler
    public void onPlayerJoin(ArenaPlayerJoinLocalEvent event) {
        if(hotbarController.isEnabled()) {
            try {
                MinigamesDTools.Companion.getInstance().getHotbarAPI().bindHotbar(this.hotbarController.buildDefaultHotbarFor(event.getPlayer()), event.getPlayer());
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
    }

    @ArenaEventHandler
    public void onPlayerLeave(ArenaPlayerLeaveLocalEvent event) {
        if(this.hotbarController.isEnabled()) {
            MinigamesDTools.Companion.getInstance().getHotbarAPI().unbindHotbar(event.getPlayer());
        }
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/gui/hotbar/HotbarListener.kt
package com.gmail.borlandlp.minigamesdtools.arena.gui.hotbar

import com.gmail.borlandlp.minigamesdtools.MinigamesDTools.Companion.instance
import com.gmail.borlandlp.minigamesdtools.arena.ArenaEventListener
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaEventHandler
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaPlayerJoinLocalEvent
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaPlayerLeaveLocalEvent
import org.bukkit.event.Listener

class HotbarListener(private val hotbarController: HotbarController) : Listener, ArenaEventListener {
    @ArenaEventHandler
    fun onPlayerJoin(event: ArenaPlayerJoinLocalEvent) {
        if (hotbarController.isEnabled) {
            try {
                instance!!.hotbarAPI!!.bindHotbar(
                    hotbarController.buildDefaultHotbarFor(
                        event.player
                    ), event.player
                )
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
    }

    @ArenaEventHandler
    fun onPlayerLeave(event: ArenaPlayerLeaveLocalEvent) {
        if (hotbarController.isEnabled) {
            instance!!.hotbarAPI!!.unbindHotbar(event.player)
        }
    }
}